espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/ComposeDriver.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.drivers
+
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.test.*
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.helpers.findAllComposeNodeInteractions
+import io.appium.espressoserver.lib.helpers.getNodeInteractionById
+import io.appium.espressoserver.lib.model.*
+
+class ComposeDriver : AppDriver {
+    override val name = DriverContext.StrategyType.COMPOSE
+
+    override fun findElement(params: Locator): BaseElement {
+        val parentNodeInteraction =
+            params.elementId?.let { getNodeInteractionById(it) }
+        val nodeInteractions =
+            findAllComposeNodeInteractions(parentNodeInteraction, params)
+
+        if (nodeInteractions.fetchSemanticsNodes().isEmpty()) {
```

Comment:
according to the docs https://developer.android.com/reference/kotlin/androidx/compose/ui/test/SemanticsNodeInteractionCollection#fetchSemanticsNodes(kotlin.Boolean,kotlin.String) this method throws an exception if atLeastOneRootRequired is set to true (which is the default value) and no matches are found

Reply:
Do you suggest it to change to
```
        try {
            nodeInteractions.fetchSemanticsNodes(true)
        } catch (e: IllegalStateException) {
            throw NoSuchElementException(
                String.format(
                    "Could not find compose element with strategy '%s' and selector '%s'",
                    params.using, params.value
                )
            )
        }
```

Reply:
You could go with the current approach, but just pass false as the first argument, otherwise an exception is going to thrown. See https://developer.android.com/reference/kotlin/androidx/compose/ui/test/SemanticsNodeInteractionCollection#fetchSemanticsNodes(kotlin.Boolean,kotlin.String): 
```
atLeastOneRootRequired: Boolean = true | Whether to throw an error in case there is no compose content in the current test app.
```

Reply:
done